packages/sidecar/src/main.ts

```diff
@@ -115,18 +121,30 @@ export function setupSidecar({
   debug,
   incomingPayload,
   isStandalone,
+  stdioMCP,
 }: SideCarOptions = {}): void {
   if (!isStandalone) {
     addEventProcessor(event => (event.spans?.some(span => span.op?.startsWith("sidecar.")) ? null : event));
   }
   let sidecarPort = DEFAULT_PORT;
 
-  if (customLogger) {
-    activateLogger(customLogger);
+  const debugLogging = debug || Boolean(process.env.SPOTLIGHT_DEBUG);
+  if (stdioMCP && !customLogger && debugLogging) {
+    logger.error("STDIO transport for MCP cannot be used with debug as they would interfere.");
+    process.exit(1);
+  } else if (debugLogging) {
+    enableDebugLogging(true);
   }
 
-  if (debug || process.env.SPOTLIGHT_DEBUG) {
-    enableDebugLogging(true);
+  if (customLogger) {
+    activateLogger(customLogger);
+  } else if (stdioMCP) {
+    activateLogger({
+      info: noop,
+      warn: noop,
+      error: noop,
+      debug: noop,
```

Comment:
not sure if this is super clever or super weird in JS.  I guess I would have done a factory function like you did with `createMCPInstance` and return a `silentLogger`